src/table/__tests__/empty-state.test.tsx

```diff
@@ -61,19 +62,29 @@ function findMergedCell(wrapper: TableWrapper) {
   return wrapper.findByClassName(styles['cell-merged']);
 }
 
+function mockResizeObserver(contentBoxWidth: number) {
+  // Use "act" to imitate firing observer's callback after component is rendered.
+  // In actual code the callback is fired twice - before and after rendering but the second part depends on ResizeObserver polyfill.
+  jest
+    .mocked(useResizeObserver)
+    .mockImplementation((_target, cb) => act(() => cb({ contentBoxWidth } as unknown as ContainerQueryEntry)));
```

Comment:
This is incorrect place for `act`. As per [the docs](https://legacy.reactjs.org/docs/testing-recipes.html#act), it is a part of "Arrange-Act-Assert", it is an action _outside_ the system under test. Putting it inside the hook mock is _inside_ the system and this is incorrect.


If you needed to handle the initial state rendering, you could make the `renderTable` async

```js
async function renderTable() {
    // render table synchronously
   return act(() => Promise.resolve());  // or another delay if you need it
}
```

Reply:
That is a hack to make the mock behave in a particular way. When in DOM the polyfill ResizeObserver fires upon creating in the use-effect which happens after the component is mounted. In the test environment that does not happen.

Reply:
Your comment does not contradict my suggestion. Why do you prefer your option?

Reply:
Did an extra dive into the same story on parallel PR: https://github.com/cloudscape-design/components/pull/1745#discussion_r1399388599

I think, both mock cases should have the same treatment